backend/dal/Services/Concrete/BuildingService.cs

```diff
@@ -358,6 +358,68 @@ public Building Update(Building building)
             return Get(existingBuilding.Id);
         }
 
+        /// <summary>
+        /// Update the specified building financials in the datasource.
+        /// </summary>
+        /// <param name="building"></param>
+        /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
+        /// <returns></returns>
+        public Building UpdateFinancials(Building building)
+        {
+            building.ThrowIfNotAllowedToEdit(nameof(building), this.User,
+                new[] {Permissions.PropertyEdit, Permissions.AdminProperties});
+            var isAdmin = this.User.HasPermission(Permissions.AdminProperties);
+
+            var existingBuilding = this.Context.Buildings
+                .Include(b => b.Evaluations)
+                .Include(b => b.Fiscals)
+                .FirstOrDefault(b => b.Id == building.Id) ?? throw new KeyNotFoundException();
+            this.ThrowIfNotAllowedToUpdate(existingBuilding, _options.Project);
+
+            var userAgencies = this.User.GetAgenciesAsNullable();
+            if (!isAdmin && !userAgencies.Contains(existingBuilding.AgencyId))
+                throw new NotAuthorizedException("User may not edit buildings outside of their agency.");
+
+            var allowEdit = isAdmin || userAgencies.Contains(existingBuilding.AgencyId);
+            if (allowEdit)
```

Comment:
could we refactor the service code so that this logic is reused? I think this is the same as what is run in the update service call right?

Reply:
I just worry about a change being made to the update but not to the financials update or vice versa.

Reply:
I agree.  The normal `Update...(...)` should probably call this function when updating the financials.  Or the financial section should be extracted into a shared extension method.